utils/nctl/sh/assets/upgrade_from_stage_single_node.sh

```diff
@@ -253,36 +253,96 @@ function _setup_asset_node_config_workaround_1()
     python3 -c "${SCRIPT[*]}"
 }
 
+function _validators_string()
+{
+    local IDX
+    local COUNT_NODES_AT_GENESIS=$(get_count_of_genesis_nodes)
+    local PATH_TO_NET
+
+    local VALIDATORS
+
+    PATH_TO_NET="$(get_path_to_net)"
+
+    for IDX in $(seq 1 "$COUNT_NODES_AT_GENESIS")
+    do
+        VALIDATORS+="--validator $(cat "$PATH_TO_NET/nodes/node-$IDX/keys/public_key_hex"),$(get_node_pos_stake_weight "$COUNT_NODES_AT_GENESIS" "$IDX") "
+    done
+
+    echo $VALIDATORS
+}
+
 function _setup_asset_global_state_toml() {
```

Comment:
this function looks the same as the one from `utils/nctl/sh/assets/setup_shared.sh`. Can we share it as not to duplicate the code?

Reply:
The part which actually takes care of the global state update tool has been deduplicated in https://github.com/casper-network/casper-node/pull/3451/commits/aec90f6309b55a11e9e50f5075a8dc3d88deccaf

If you're not strong about it, I'd like to not touch the prelude of these functions until we have the upgrade scenario working correctly, so I can be sure that refactorings made to the tests don't break things.